### ISSUE TITLE
Throws System.ArgumentException while reading a local NuGet.config file

### ISSUE THREAD

**Issue by msschl** [NuKeeperDotNet/NuKeeper#439]
#### Steps to reproduce
- Create a new project
- Place a NuGet.config file in the project root directory with the follwing content:
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <clear />
    <!-- Restore sources should be defined in build/sources.props. -->
  </packageSources>
</configuration>
```
- Run the command `nukeeper inspect -v d`

#### Produces
```shell
Reading file F:\Programming\c#\Geocoding\NuGet.Config for package sources
Reading file C:\Users\XXX\AppData\Roaming\NuGet\NuGet.Config for package sources
Reading file C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config for package sources

Unhandled Exception: System.ArgumentException: sources
   at NuKeeper.Inspection.Sources.NuGetSources..ctor(IEnumerable`1 sources) in C:\Code\NuKeeper\NuKeeper.Inspection\Sources\NuGetSources.cs:line 38
   at NuKeeper.Inspection.Sources.NuGetConfigFileReader.ReadFromFile(IReadOnlyCollection`1 sources) in C:\Code\NuKeeper\NuKeeper.Inspection\Sources\NuGetConfigFileReader.cs:line 41
   at NuKeeper.Inspection.Sources.NuGetConfigFileReader.ReadNugetSources(IFolder workingFolder) in C:\Code\NuKeeper\NuKeeper.Inspection\Sources\NuGetConfigFileReader.cs:line 30
   at NuKeeper.Inspection.Sources.NuGetSourcesReader.Read(IFolder workingFolder, NuGetSources overrideValues) in C:\Code\NuKeeper\NuKeeper.Inspection\Sources\NuGetSourcesReader.cs:line 26
   at NuKeeper.Local.LocalEngine.Run(SettingsContainer settings, Boolean write) in C:\Code\NuKeeper\NuKeeper\Local\LocalEngine.cs:line 43
   at NuKeeper.Commands.InspectCommand.Run(SettingsContainer settings) in C:\Code\NuKeeper\NuKeeper\Commands\InspectCommand.cs:line 33
   at NuKeeper.Commands.CommandBase.OnExecute() in C:\Code\NuKeeper\NuKeeper\Commands\CommandBase.cs:line 69
   at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.InvokeAsync(MethodInfo method, Object instance, Object[] arguments) in C:\projects\commandlineutils\src\CommandLineUtils\Conventions\ExecuteMethodConvention.cs:line 77
   at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.OnExecute(ConventionContext context) in C:\projects\commandlineutils\src\CommandLineUtils\Conventions\ExecuteMethodConvention.cs:line 62
   at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.<>c__DisplayClass0_0.<<Apply>b__0>d.MoveNext() in C:\projects\commandlineutils\src\CommandLineUtils\Conventions\ExecuteMethodConvention.cs:line 25
--- End of stack trace from previous location where exception was thrown ---
   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.<>c__DisplayClass126_0.<OnExecute>b__0() in C:\projects\commandlineutils\src\CommandLineUtils\CommandLineApplication.cs:line 505
   at NuKeeper.Program.Main(String[] args) in C:\Code\NuKeeper\NuKeeper\Program.cs:line 31
```

### Suggestion
Even when the user clears all his packageSources in the NuGet.config file we should not throw an exception like this.

**Comment by AnthonySteele** [comment]
This is an edge case, but yes, we should not crash.
So NuKeeper should treat an empty list of sources the same as a missing list, and proceed with fallback, probably on to using the public global feed.  The alternative of looking for updates from 0 sources doesn't make sense.

**Comment by msschl** [comment]
You are right. Updates from 0 sources doesn't make any sense.

Unfortunately my project is very special and complex. I have CI/CD integration and seperated all package versions out in a dependencies.props file. My project structure looks something like this:

![grafik](https://user-images.githubusercontent.com/5054780/45741352-a7b87500-bbf7-11e8-99ec-2c2f6d67eb14.png)

I am searching for an tool which can handle such an edge case. While testing this tool I discoverd this exception.

**Comment by AnthonySteele** [comment]
NuKeeper will eventually run standard `dotnet add` commands, and as far as I have seen, this will only update the `.csproj` files, not the custom ones.

**Comment by msschl** [comment]
I have looked into the source code of the dotnet add command and in the source code of a tool called dotnet outdated. Both uses msbuild internally to create a restore graph. Unfortunately the restore graph does not include the location to the specific `.props` file from where the package was referenced or the version specified.

I created an [issue](https://github.com/Microsoft/msbuild/issues/3770) on the [Microsoft/msbuild](https://github.com/Microsoft/msbuild) repo. If they decide to add this feature tools like dotnet add and dotnet outdated can use this and even cover this edge case.
